archive/2025-11-07-what-greynoise-learned-from-deploying-mcp-honeypots.md

```diff
@@ -0,0 +1,211 @@
+
+[![View our Bluesky profile](https://cdn.prod.website-files.com/667dd40ebb8095e89f275639/679bf65fd61237e62cf27065_bluesky-icon.svg)](https://bsky.app/intent/compose?text=What%20GreyNoise%20Learned%20from%20Deploying%20MCP%20Honeypots%20https%3A%2F%2Fwww.greynoise.io%2Fblog%2Fdeploying-mcp-honeypots)[![View our profile on X](https://cdn.prod.website-files.com/667dd40ebb8095e89f275639/667dd40ebb8095e89f275b13_twitter-logo.png)](https://twitter.com/intent/tweet?text=What%20GreyNoise%20Learned%20from%20Deploying%20MCP%20Honeypots&url=https%3A%2F%2Fwww.greynoise.io%2Fblog%2Fdeploying-mcp-honeypots)[![View our LinkedIn profile](https://cdn.prod.website-files.com/667dd40ebb8095e89f275639/667dd40ebb8095e89f275a81_linkedin-icon.svg)](https://www.linkedin.com/sharing/share-offsite/?url=https%3A%2F%2Fwww.greynoise.io%2Fblog%2Fdeploying-mcp-honeypots)
+
+![](https://cdn.prod.website-files.com/667dd40ebb8095e89f27565d/6905335bb5a4365c300b0444_GreyNoise-MCP-Honeypot-insights-blog-1600x900.png)
+
+The Model Context Protocol (MCP) lets AI systems securely connect to external data and tools — a key building block for emerging AI-driven operations. It’s also a new piece of internet-facing infrastructure, which means defenders are asking the same question they ask of anything exposed online: _is anyone attacking it yet?_
+
+To find out, **GreyNoise deployed a series of MCP honeypots to observe what actually happens when AI middleware meets the open internet.**
+
+‍
+
+## Testing the Exposure
+
+Each honeypot mimicked a plausible MCP deployment, instrumented for full packet-level visibility. We built three configurations:
+
+- **Unauthenticated endpoint** to capture background scanning.
+- **Authenticated endpoint** requiring an API key to detect credential probing.
+- **Simulated developer instance** with a deliberately exposed key to see whether anyone followed it.
+
+All were isolated from production systems.
+
+**Every instance was discovered within days** — proof that anything listening on the internet will be found quickly. After that, activity leveled off. We saw the usual: HTTP probes, SSH touches, and other one-shot scans indistinguishable from the constant hum that hits everything else online.
+
+Across the deployment, **no MCP-specific payload or exploitation attempts appeared.**
+
+GreyNoise’s broader MCP Scanner tag, which tracks reconnaissance against MCP endpoints, shows that threat actors are indeed discovering exposed MCP servers — and increasingly so.
+
+‍
+
+## A Controlled Exception
+
+In October 2025, independent researchers [demonstrated](https://www.theregister.com/2025/10/21/mcp_prompt_hijacking_attack/) a prompt-hijacking flaw in a custom MCP build that used a deprecated protocol handler. It was a contained proof-of-concept, not an attack in the wild — and it supports the same conclusion: present-day MCP exposure risk lies in implementation errors, not in deliberate targeting of MCPs as a class.
```

Comment:
Deprecated protocol makes it sound like a patched vulnerability. Plenty of case studies were demonstrated by the hacker community. MCP is literally using prompt injection as a feature..